File: CSCI24000_fall2020_A7-master/base/Account.java
// Account.java
// This is the Account class. Each account has type, balance, interest rate and withdraw, deposit, showBalance methods

import java.util.*;

public class Account{
	String type;
	double balance;
	double interestRate;

	public Account(String x,double y,double z){
		type = x;
		balance = y;
		interestRate = z;
	} // end constructor

	public static void main(String args[]){
		Account savings = new Account("hi,1000,1");
		System.out.println("Is the amount below a thousand?");
		savings.showBalance();
	} // end main
	
	public Account(String properties){
		String[] props = new String[3];	
		for(int i=0;i<3;i++){
			int index = properties.indexOf(',');
			if(index<0){
				props[i]=properties;
				break;
			}
			props[i] = properties.substring(0,index);
			properties = properties.substring(index+1);
		}
		type = props[0];
		balance = Double.parseDouble(props[1]);
		interestRate = Double.parseDouble(props[2]);

	}; // end constructor

	public String getProps(){
		return type+","+balance+","+interestRate;
	}

	public void withdraw(double withdrawAmount){
		if(withdrawAmount>balance){
			System.out.println("Insufficient funds");
		}
		else
			balance = balance - withdrawAmount;
	} // end withdraw

	public void deposit(double depositAmount){
		balance = balance + depositAmount;
	} // end deposit

	public String getType(){
		return type;
	}
	public double getBalance(){
		return balance;
	}

	public double getInterestRate(){
		return interestRate;
	}

	public void showBalance(){
		System.out.println(balance);
	}
} // end class def

File: CSCI24000_fall2020_A7-master/blackbelt/account.cpp
// Account.java
// This is the Account class. Each account has type, balance, interest rate and withdraw, deposit, showBalance methods

#include "account.h"
#include <string>
#include <iostream>
#include <sstream>
    
Account::Account(std::string x,double y,double z){
		Account::type = new std::string(x);
		Account::balance = new double(y);
		Account::interestRate = new double(z);
	} // end constructor

	Account::Account(std::string properties){
		std::string props[3];	
		for(int i=0;i<3;i++){
			int index = properties.find(',');
			if(index<0){
				props[i]=properties;
				break;
			}
			props[i] = properties.substr(0,index);
			properties = properties.substr(index+1);
		}
		Account::type = new std::string(props[0]);
        double b;
        std::stringstream balance(props[1]);
        balance>>b;
		Account::balance = new double(b);
        double i;
        std::stringstream interest(props[2]);
        interest>>i;
		Account::interestRate = new double(i);

	}; // end constructor

    Account::~Account(){
        delete type;
        delete balance;
        delete interestRate;
    }

	std::string Account::getProps(){
        std::stringstream ss;
        ss << *type<<","<<*balance<<","<<*interestRate;
		return ss.str();
	}

	void Account::withdraw(double withdrawAmount){
		if(withdrawAmount>*balance){
			std::cout<<"Insufficient funds"<<std::endl;
		}
		else
			balance = new double(*balance - withdrawAmount);
	} // end withdraw

	void Account::deposit(double depositAmount){
		balance = new double(*balance + depositAmount);
	} // end deposit

	std::string Account::getType(){
		return *type;
	}
	double Account::getBalance(){
		return *balance;
	}

	double Account::getInterestRate(){
		return *interestRate;
	}

	void Account::showBalance(){
		std::cout<<*balance<<std::endl;
	}

